artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ResetableIterator.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.server.impl;
+
+import java.util.Iterator;
+
+public interface ResetableIterator<T> extends Iterator<T> {
+
+   /**
+    * Resets the iterator so you can re-iterate over all elements.
+    *
+    * @return itself, this is just for convenience.
+    */
+   ResetableIterator<T> reset();
```

Comment:
We can extends `AutoCloseable` too and override `close`  in order to call `reset` by default too

Reply:
-1 we are not closing the iterator, nor would this go in a try resources block., we are simply resetting the iterator so it marks the endpos = startpos, so we continue to round robin, as successfully handled a message.

Reply:
ill mark resolved then.